drivers/modem/modem_cellular.c

```diff
@@ -1560,6 +1581,32 @@ static int modem_cellular_init(const struct device *dev)
 							&dlci2_config);
 	}
 
+	{
+		const struct modem_cmux_dlci_config dlci3_config = {
+			.dlci_address = 3,
+			.receive_buf = data->dlci2_receive_buf,
```

Comment:
```suggestion
			.receive_buf = data->dlci3_receive_buf,
```
Having those in an array that is looped over could be more handy (and prevent such errors).

Reply:
Agreed, I'm working on a proper sample that creates a bunch of DLCI channels for pipelinks, I'm doing the following to prevent these errors:
```
struct bg9x_pipe {
	struct modem_cmux_dlci dlci;
	uint8_t dlci_address;
	uint8_t *dlci_receive_buf;
	uint16_t dlci_receive_buf_size;
	struct modem_pipe *pipe;
	struct modem_pipelink *pipelink;
};

static void bg9x_init_pipe(struct modem_cmux *cmux, struct bg9x_pipe *pipe)
{
	struct modem_cmux_dlci dlci = &pipe->dlci;
	const struct modem_cmux_dlci_config config = {
		.dlci_address = pipe->dlci_address,
		.receive_buf = pipe->dlci_receive_buf,
		.receive_buf_size = pipe->dlci_receive_buf_size,
	};

	pipe->pipe = modem_cmux_dlci_init(cmux, dlci, &config);

	if (pipe->pipelink != NULL) {
		modem_pipelink_init(pipe->pipelink, pipe->pipe);
	}
}

static void bg9x_init_pipes(const struct device *dev)
{
	const struct bg9x_config *config = dev->config;
	struct bg9x_data *data = dev->data;
	struct modem_cmux *cmux;
	struct bg9x_pipe *pipe;

	for (size_t i = 0; i < config->dlcis_size; i++) {
		cmux = &data->cmux;
		pipe = &config->pipes[i];
		bg9x_init_pipe(cmux, pipe);
	}
}
```
Its only an snippet of the actual sample I'm working on